src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2624,7 +2624,51 @@ void ApplicationManagerImpl::UnregisterApplication(
     } else {
       resume_controller().RemoveApplicationFromSaved(app_to_remove);
     }
+
+    application_manager::VehicleInfoSubscriptions vehicle_data_for_unsubscribe =
+        SelectVehicleDataForUnsubscribe(app_to_remove);
+
+    if (!vehicle_data_for_unsubscribe.empty()) {
+      auto message_to_hmi =
+          MessageHelper::CreateUnsubscribeVehicleDataMessageForHMI(
+              vehicle_data_for_unsubscribe, app_to_remove);
+
+#ifdef DEBUG
+      MessageHelper::PrintSmartObject(*message_to_hmi);
+#endif
+
+      CommandSharedPtr command =
+          HMICommandFactory::CreateCommand(message_to_hmi, *this);
+      if (!command) {
+        LOG4CXX_WARN(logger_, "Failed to create command from smart object");
+      }
```

Comment:
@IGapchuk  i think there should be a return statement here, otherwise command->Init() or command->Run() below may result in a crash

Reply:
That changes was revert.